src/rule_definition_tools.jl

```diff
@@ -165,7 +165,7 @@ function scalar_frule_expr(__source__, f, call, setup_stmts, inputs, partials)
     return @strip_linenos quote
         # _ is the input derivative w.r.t. function internals. since we do not
         # allow closures/functors with @scalar_rule, it is always ignored
-        function ChainRulesCore.frule((_, $(Δs...)), ::typeof($f), $(inputs...))
+        function ($ChainRulesCore.frule)(($(esc(:_)), $(Δs...)), ::$(typeof)($f), $(inputs...))
```

Comment:
No need to escape `_`

Reply:
As mentioned in https://github.com/JuliaDiff/ChainRulesCore.jl/pull/325/files#r604881501, the main intention was to obtain a (somewhat) cleaner output. I don't mind changing it though.

Reply:
Seems like `_` getting mangled by macro hygiene is something that we should fix in Base. It definitely shouldn't get turned into a `GlobalRef`. Mind opening an issue?

Reply:
Ok, https://github.com/JuliaLang/julia/pull/40280 should fix this